src/java.base/share/classes/java/lang/foreign/SegmentScope.java

```diff
@@ -0,0 +1,107 @@
+package java.lang.foreign;
+
+import jdk.internal.foreign.MemorySessionImpl;
+import jdk.internal.javac.PreviewFeature;
+import jdk.internal.ref.CleanerFactory;
+
+/**
+ * A segment scope controls access to a memory segment.
```

Comment:
A passing comment here is that "to a memory segment" hints of one-to-one relationship when it's actually one-to-many. Arena is specified to control the lifecycle "of memory segments".